allennlp/tests/nn/util_test.py

```diff
@@ -681,6 +683,106 @@ def test_sequence_cross_entropy_with_logits_averages_token_correctly(self):
         average_token_loss = (total_token_loss / weights.float().sum()).detach()
         assert_almost_equal(loss.detach().item(), average_token_loss.item())
 
+    def test_sequence_cross_entropy_with_logits_gamma_correctly(self):
+        batch = 1
+        length = 3
+        classes = 4
+        gamma = abs(numpy.random.randn()) # [0, +inf)
+
+        tensor = torch.rand([batch, length, classes])
+        targets = torch.LongTensor(numpy.random.randint(0, classes, [batch, length]))
+        weights = torch.ones([batch, length])
+
+        loss = util.sequence_cross_entropy_with_logits(tensor, targets, weights, gamma=gamma)
+
+        correct_loss = 0.
+        for logit, label in zip(tensor.squeeze(0), targets.squeeze(0)):
+            p = torch.nn.functional.softmax(logit, dim=-1)
+            pt = p[label]
+            correct_loss += pt.log() * (1 - pt) ** gamma
+        # Average over sequence.
+        correct_loss = - correct_loss / length
```

Comment:
Slightly clearer to subtract inside the loop and just average here.  Same for all tests below.

Reply:
Great. Thanks.